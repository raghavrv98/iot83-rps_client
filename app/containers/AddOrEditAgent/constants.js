/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
