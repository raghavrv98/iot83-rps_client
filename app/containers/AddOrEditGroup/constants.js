/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
