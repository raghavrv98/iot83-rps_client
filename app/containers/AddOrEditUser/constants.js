/*
 *
 * AddOrEditUser constants
 *
 */

export const DEFAULT_ACTION = "app/AddOrEditUser/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddOrEditUser/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditUser/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditUser/ON_SUBMIT_REQUEST_FAILURE";

export const GET_USER_DETAILS = "app/AddOrEditUser/GET_USER_DETAILS";
export const GET_USER_DETAILS_SUCCESS = "app/AddOrEditUser/GET_USER_DETAILS_SUCCESS";
export const GET_USER_DETAILS_FAILURE = "app/AddOrEditUser/GET_USER_DETAILS_FAILURE";

export const GET_ALL_ROLES = "app/AddOrEditUser/GET_ALL_ROLES";
export const GET_ALL_ROLES_SUCCESS = "app/AddOrEditUser/GET_ALL_ROLES_SUCCESS";
export const GET_ALL_ROLES_FAILURE = "app/AddOrEditUser/GET_ALL_ROLES_FAILURE";

export const GET_ALL_GROUPS = "app/AddOrEditUser/GET_ALL_GROUPS";
export const GET_ALL_GROUPS_SUCCESS = "app/AddOrEditUser/GET_ALL_GROUPS_SUCCESS";
export const GET_ALL_GROUPS_FAILURE = "app/AddOrEditUser/GET_ALL_GROUPS_FAILURE";
