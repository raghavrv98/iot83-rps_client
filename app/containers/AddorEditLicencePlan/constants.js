/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";