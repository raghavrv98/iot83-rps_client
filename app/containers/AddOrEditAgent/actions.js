/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
