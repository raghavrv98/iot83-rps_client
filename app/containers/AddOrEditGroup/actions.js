/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
