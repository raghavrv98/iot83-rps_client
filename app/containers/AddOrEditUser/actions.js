/*
 *
 * AddOrEditUser actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_USER_DETAILS, GET_ALL_GROUPS, GET_ALL_ROLES } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload, id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getUserDetails(id) {
  
  return {
    type: GET_USER_DETAILS,
    id
  };
}

export function getAllRole() {
  return {
    type: GET_ALL_ROLES
  }
}

export function getAllGroup() {
  return {
    type: GET_ALL_GROUPS
  }
}