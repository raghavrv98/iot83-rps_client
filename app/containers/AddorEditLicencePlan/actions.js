/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}