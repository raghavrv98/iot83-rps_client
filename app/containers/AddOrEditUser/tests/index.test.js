import React from 'react';
import { shallow } from 'enzyme';
import history from 'utils/history';
import {onSubmitHandler, getAllGroup, getAllRole, getUserDetails} from '../actions'
import { AddOrEditUser, mapDispatchToProps } from '../index';
import { ON_SUBMIT_REQUEST, GET_ALL_GROUPS, GET_ALL_ROLES, GET_USER_DETAILS } from "../constants";

let mockParams = {
  params: {
    id: ""
  }
}

const submitSuccess =  "this is submit success";
const submitError =  "this is submit error";
const userDetails =  {
  company: "pmmp ",
  compartments: ["sampleGroupId"],
  email: "",
  firstName: "",
  gender: "",
  lastName: "",
  mobilePhone: "",
  roles: ["sampleRoleId"],
  title: ""
}
const userDetailsError =  "this is user details error";
const roles =  [{id: "sampleRoleId", name: "sampleRoleName"}];
const rolesError =  "this is submit error";
const groups =  [{id: "sampleGroupId", name: "sampleGroupName"}];
const groupsError =  "this is submit error";

describe('<AddOrEditUser />', () => {

  // it("clicking back button trigger onClick event", async () => {
  //   let expectedCount = history.length + 1
  //   const wrapper = shallow(<AddOrEditUser history={history} onSubmitHandler={onSubmitHandler} match={mockParams} getGroups={getAllGroup} getRoles={getAllRole} />);
  //   let backbutton = wrapper.find("button.btn-transparent");
  //   expect(backbutton.length).toBe(1);
  //   backbutton.simulate('click');
  //   expect(history.length).toEqual(expectedCount);
  //   expect(history.location.pathname).toEqual("/manageUsers");
  // });

  // it("clicking submit button trigger onClick event", async () => {
  //   const onSubmitHandlerSpy = jest.spyOn(AddOrEditUser.prototype, "onSubmitHandler");
  //   const wrapper = shallow(<AddOrEditUser history={history} onSubmitHandler={onSubmitHandler} match={mockParams} getGroups={getAllGroup} getRoles={getAllRole} />);
  //   let loaderDiv = wrapper.find("div.pageLoaderBox");
  //   expect(loaderDiv.length).toBe(1);
  //   wrapper.setProps({ submitSuccess, submitError, userDetails, userDetailsError, roles, rolesError, groups, groupsError });
  //   let addOrEditForm = wrapper.find("form.contentForm");
  //   expect(addOrEditForm.length).toBe(1);
  //   addOrEditForm.simulate('submit', { preventDefault() { } });
  //   expect(onSubmitHandlerSpy).toHaveBeenCalledTimes(1);
  // });

  it("test component will recieve props", async () => {
    let expectedCount = history.length + 1
    spyOn(AddOrEditUser.prototype, 'componentWillReceiveProps').and.callThrough();
    const wrapper = shallow(<AddOrEditUser history={history} match={mockParams} getGroups={getAllGroup} getRoles={getAllRole} onSubmitHandler={onSubmitHandler} submitSuccess="" submitError="" userDetails="" userDetailsError="" roles="" rolesError="" groups="" groupsError="" />);
    wrapper.setProps({ submitSuccess, submitError, userDetails, userDetailsError, roles, rolesError, groups, groupsError });
    // expect(history.length).toEqual(expectedCount);
    // expect(history.location.pathname).toEqual("/manageUsers");
    expect(AddOrEditUser.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
  });

  it("test edit case", async () => {
    const mockParams = {
      params: {
        id: "demoId"
      }
    }
    spyOn(AddOrEditUser.prototype, 'componentDidMount').and.callThrough();
    const wrapper = shallow(<AddOrEditUser history={history} match={mockParams} getUserDetails={getUserDetails} getGroups={getAllGroup} getRoles={getAllRole} />);
    expect(AddOrEditUser.prototype.componentDidMount).toHaveBeenCalledTimes(1);
  });

  it('match snapshots', () => {
    const wrap = shallow(<AddOrEditUser history={history} onSubmitHandler={onSubmitHandler} match={mockParams} getGroups={getAllGroup} getRoles={getAllRole} />)
    expect(wrap).toMatchSnapshot()
  })

  // test('Input should call handleChange on name change event', () => {
  //   const wrapper = shallow(<AddOrEditUser history={history} match={mockParams} getUserDetails={getUserDetails} getGroups={getAllGroup} getRoles={getAllRole}  />);
  //   const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), 'inputChangeHandler');
  //   wrapper.instance().forceUpdate();
  //   let loaderDiv = wrapper.find("div.pageLoaderBox");
  //   expect(loaderDiv.length).toBe(1);
  //   wrapper.setProps({ submitSuccess, submitError, userDetails, userDetailsError, roles, rolesError, groups, groupsError });
  //   wrapper.find('[id="firstName"]').simulate('change',{
  //     target: {id: "firstName", value: "name"}
  //   });
  //   expect(wrapper.state().payload.firstName).toBe('name');
  //   expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
  // })
  test('Input should call handleChange on role change event', () => {
    const wrapper = shallow(<AddOrEditUser history={history} match={mockParams} getUserDetails={getUserDetails} getGroups={getAllGroup} getRoles={getAllRole}  />);
    wrapper.setProps({roles, rolesError, groups, groupsError });
    const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), 'inputChangeHandler');
    wrapper.instance().forceUpdate();
    wrapper.find('[id="roles"]').simulate('change',{
      target: {id: "roles", value: "sampleRoleId"}
    });
    expect(wrapper.state().payload.roles).toEqual(["sampleRoleId"]);
    expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
  })
  it('Function testing on mapDispatchToProps', () => {
    const dispatch = jest.fn();
    const payload = "demo"
    const mockParams = {
      params: {
        id: "demoId"
      }
    }
    mapDispatchToProps(dispatch).onSubmitHandler(payload,mockParams.params.id);
    expect(dispatch.mock.calls[0][0]).
    toEqual({ 
      type: ON_SUBMIT_REQUEST,
      payload: 'demo',
      id: 'demoId' 
    })
    mapDispatchToProps(dispatch).getUserDetails(mockParams.params.id);
    expect(dispatch.mock.calls[1][0]).
    toEqual({
      type: GET_USER_DETAILS,
      id: "demoId"
    })
    mapDispatchToProps(dispatch).getGroups();
    expect(dispatch.mock.calls[2][0]).
    toEqual({
      type: GET_ALL_GROUPS
    })
    mapDispatchToProps(dispatch).getRoles();
    expect(dispatch.mock.calls[3][0]).
    toEqual({
      type: GET_ALL_ROLES
    })
  });
});
