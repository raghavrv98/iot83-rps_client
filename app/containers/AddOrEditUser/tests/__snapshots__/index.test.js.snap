// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddOrEditUser /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      AddOrEditUser
    </title>
    <meta
      content="Description of AddOrEditUser"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p
        className="pd-l-30"
      >
        <span
          className="cursor-pointer"
          onClick={[Function]}
        >
          <button
            className="btn btn-transparent"
          >
            <i
              className="far fa-long-arrow-left"
            />
          </button>
          <FormattedMessage
            defaultMessage="Manage Users"
            id="app.containers.AddOrEditUser.title"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
        </span>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="Create User"
          id="app.containers.AddOrEditUser.titleCreateUser"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
    <div
      className="flex-item fx-b30 text-right align-items-center"
    />
  </div>
  <SkeletonLoader
    mode="fullView"
    skeleton="skeletonNestedFormUpload"
  />
</div>
`;
